ui_text_input.js

```diff
@@ -0,0 +1,34 @@
+module.exports = function(RED) {
+    var ui = require('../ui')(RED);
+
+    function TextInputNode(config) {
+        RED.nodes.createNode(this, config);
+        var node = this;
+
+        var group = RED.nodes.getNode(config.group);
+        if (!group) { return; }
+        var tab = RED.nodes.getNode(group.config.tab);
+        if (!tab) { return; }
+
+        var done = ui.add({
+            node: node,
+            tab: tab,
+            group: group,
+            control: {
+                type: config.delay < 0 ? 'text-input-CR' : 'text-input',
```

Comment:
when 0 is set, currently each key stroke gets passed through immediately.
So I considered <0 makes sense...